include/CMakeLists.txt

```diff
@@ -1 +1,43 @@
-install(DIRECTORY yamail DESTINATION include)
+find_package(Boost 1.66 REQUIRED COMPONENTS system thread REQUIRED)
+
+add_library(resource_pool INTERFACE)
+target_compile_features(resource_pool INTERFACE cxx_std_14)
+target_include_directories(resource_pool
+        INTERFACE
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+        $<INSTALL_INTERFACE:include>)
+target_link_libraries(resource_pool INTERFACE Boost::system Boost::thread Boost::boost)
+target_compile_definitions(resource_pool
+        INTERFACE
+        -DBOOST_COROUTINES_NO_DEPRECATION_WARNING
+        )
+add_library(yandex::resource_pool ALIAS resource_pool)
+
+install(TARGETS resource_pool EXPORT resource_poolTargets
+        LIBRARY DESTINATION lib COMPONENT Runtime
+        ARCHIVE DESTINATION lib COMPONENT Development
+        RUNTIME DESTINATION bin COMPONENT Runtime
+        PUBLIC_HEADER DESTINATION include COMPONENT Development
+        BUNDLE DESTINATION bin COMPONENT Runtime)
+install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} DESTINATION .)
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file("${CMAKE_CURRENT_BINARY_DIR}/resource_pool/resource_poolConfigVersion.cmake"
+        COMPATIBILITY AnyNewerVersion)
+
+export(EXPORT resource_poolTargets
+        FILE "${CMAKE_CURRENT_BINARY_DIR}/resource_pool/resource_poolTargets.cmake"
+        NAMESPACE yandex::)
```

Comment:
Using yandex namespace is not very appropriate. Change to elsid please.

Reply:
Sorry, I didn't mean to insult you or discredit your work. As stated, this work started in the depths of ozo'z `contrib/`, so I just copied what I had there.
I have changed it now to properly reflect ownership.

Reply:
No problem, I wasn't insulted. It's moslty not about the ownership, but just about where this repository is located.